### PR TITLE
redis: adding the missing `notify-keyspace-events` value

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Whilst this value can currently be defined via the additionalProperties in the Track1 Go SDK, since it's included in the description presumably this should also be an explicitly defined field? Description:

> "description": "All Redis Settings. Few possible keys: rdb-backup-enabled,rdb-storage-connection-string,rdb-backup-frequency,maxmemory-delta,maxmemory-policy,notify-keyspace-events,maxmemory-samples,slowlog-log-slower-than,slowlog-max-len,list-max-ziplist-entries,list-max-ziplist-value,hash-max-ziplist-entries,hash-max-ziplist-value,set-max-intset-entries,zset-max-ziplist-entries,zset-max-ziplist-value etc."

As it stands today this field can be set via adding this to/parsing this from the dictionary, however since other properties have been defined explicitly, presumably this one should be too?

More details:

* [Azure Redis blog post](https://techcommunity.microsoft.com/t5/azure-paas-blog/redis-keyspace-events-notifications/ba-p/1551134)
* [Redis documentation](https://redis.io/docs/manual/keyspace-notifications/)

### CODE REVIEW

File: specification/redis/resource-manager/Microsoft.Cache/stable/2021-06-01/redis.json
Comment (TimLovellSmith) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2021-06-01/redis.json:1938]: >The KeySpace Events [](http://example.com/codeflow?start=30&length=19)

```suggestion
the keyspace events
```
doc style capitalization

**Review by TimLovellSmith — APPROVED** [review]
:shipit:

**Comment by TimLovellSmith** [comment]
Technically we could also have a validation regex for allowed values, but that's something that depends on the redis version, which will be hard to model and keep up to date...

File: specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/redis.json
Comment (TimLovellSmith) [inline specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/redis.json:1952]: Doc capitalization can be as ordinary english

Comment (tombuildsstuff) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/redis.json:1952]: thanks for the review - to clarify, do you mean?

```suggestion
              "description": "The keyspace events which should be monitored."
```

Comment (TimLovellSmith) [reply specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/redis.json:1952]: Yup, that works

**Review by TimLovellSmith — APPROVED** [review]

**Comment by tombuildsstuff** [comment]
@dw511214992 since this has been signed off, what's needed to get this merged?

**Comment by dw511214992** [comment]
@tombuildsstuff @TimLovellSmith Why was the label `NewApiVersionRequired` removed?

**Comment by TimLovellSmith** [comment]
@dw511214992 My thinking is that because its customer reported bugfix to the swagger for this api-version, it can't be in a new api-version.

**Comment by tombuildsstuff** [comment]
I've updated this PR to apply this fix to the two most recent API Versions (`2023-05-01-preview` and `2023-08-01`)

**Comment by tombuildsstuff** [comment]
@dw511214992 @lirenhe @TimLovellSmith what's remaining to get this one merged? From what I can see this is now good to go?

**Comment by tombuildsstuff** [comment]
@rkmanda ping https://github.com/Azure/azure-rest-api-specs/pull/22407#issuecomment-1807865492?

**Comment by TimLovellSmith** [comment]
Sory for the lack of progress. I think we're all happy with this as a bugfix to docs from an ARM perspective, its just breaking change / versioning analysis needed.

**Comment by tombuildsstuff** [comment]
@TimLovellSmith given there's a new version of the Redis API on a frequent enough basis, if it's easier to add support for this to the new API Version and then just focus on that new version, that's also fine with me fwiw - backporting this to previous versions is mostly to ensure the Swagger definitions are correct, more than anything.

**Comment by TimLovellSmith** [comment]
@JeffreyRichter Do you think this contribution is okay as is, as a breaking change bugfix?

**Comment by TimLovellSmith** [comment]
@konrad-jamrozik Who does versioning review?

**Comment by konrad-jamrozik** [comment]
> @konrad-jamrozik Who does versioning review?

Per https://aka.ms/azsdk/pr-diagram this is handled by Breaking Change Review Board (step 1).

**Comment by JeffreyRichter** [comment]
> @JeffreyRichter Do you think this contribution is okay as is, as a breaking change bugfix?

If the service versions actually supported this over the wire but it was missing from the swagger, then we can approve the breaking change as a "BugFix".

**Comment by TimLovellSmith** [comment]
> > @JeffreyRichter Do you think this contribution is okay as is, as a breaking change bugfix?
> 
> If the service versions actually supported this over the wire but it was missing from the swagger, then we can approve the breaking change as a "BugFix".

Yes, that is the case.

**Comment by konrad-jamrozik** [comment]
@tombuildsstuff this PR is from over a year ago so the automation didn't add `Next Steps to Merge` comment. But you can see what needs to be done to unblock this PR here in the `Automated merging requirements met` check page:

- https://github.com/Azure/azure-rest-api-specs/pull/22407/checks?check_run_id=23860242916

As well as at https://aka.ms/azsdk/pr-diagram

I just ensured your PR will be on [ARM review queue](https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests). Looks like your PR is first. Hence ARM on-call should review this PR soon and provide feedback or approve.

**Pull request merged**
